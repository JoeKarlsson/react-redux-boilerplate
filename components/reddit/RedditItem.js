'use strict';

import React from 'react';
import { connect } from 'react-redux'
import PropTypes from 'prop-types';
import { removeItem } from '../../actions/redditActions';

class RedditItem extends React.Component {
  constructor() {
    super();
    this.handleDelete = this.handleDelete.bind(this);
  };

  handleDelete(e) {
    const id = e.target.id;
    const { dispatch, index } = this.props;
    dispatch(removeItem(index));
  };

  render() {
    return (
      <div className='redditItem'>
        <h3>{ this.props.children }</h3>
        <p>Author: {this.props.author}</p>
        <button
          id={this.props.id}
          onClick={this.handleDelete} >
          X
        </button>
      </div>
    )
  }
};

const mapStateToProps = (state) => {
  return { }
};

export default connect(
  mapStateToProps
)(RedditItem);
