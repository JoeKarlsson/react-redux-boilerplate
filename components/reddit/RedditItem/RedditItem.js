import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { removeItem } from '../../../actions/redditActions';
import styles from './RedditItem.scss';

class RedditItem extends React.Component {
  constructor() {
    super();
    this.handleDelete = this.handleDelete.bind(this);
  }

  handleDelete() {
    const { dispatch, index } = this.props;
    dispatch(removeItem(index));
  }

  render() {
    return (
      <div className={styles.RedditItem}>
        <h3>{ this.props.children }</h3>
        <p>Author: {this.props.author}</p>
        <button
          id={this.props.id}
          onClick={this.handleDelete}
        >
          X
        </button>
      </div>
    );
  }
}

RedditItem.propTypes = {
  author: PropTypes.string,
  children: PropTypes.string,
  dispatch: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
};

RedditItem.defaultProps = {
  redditData: [],
  author: 'Default',
  children: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorum dolorem architecto eos, esse facere quo omnis expedita, impedit officiis! Quisquam officiis recusandae expedita numquam vel in molestias qui placeat esse.'
};

export default connect()(RedditItem);
