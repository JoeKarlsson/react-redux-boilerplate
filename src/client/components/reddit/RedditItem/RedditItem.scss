$purple: purple;

.RedditItem {
	background-color: $purple;
}
