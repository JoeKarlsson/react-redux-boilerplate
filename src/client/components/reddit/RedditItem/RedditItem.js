import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { removeItem } from '../../../actions/redditActions';
import './RedditItem.scss';

export class RedditItem extends React.Component {
	constructor() {
		super();
		this.handleDelete = this.handleDelete.bind(this);
	}

	handleDelete() {
		const { dispatch, index } = this.props;
		dispatch(removeItem(index));
	}

	render() {
		return (
			<div className="RedditItem">
				<h3>{ this.props.children }</h3>
				<p>Author: {this.props.author}</p>
				<button
					onClick={this.handleDelete}
				>
          X
				</button>
			</div>
		);
	}
}

RedditItem.propTypes = {
	author: PropTypes.string.isRequired,
	index: PropTypes.number.isRequired,
	children: PropTypes.string,
	dispatch: PropTypes.func.isRequired,
};

RedditItem.defaultProps = {
	children: '',
};

export default connect(null, null)(RedditItem);
