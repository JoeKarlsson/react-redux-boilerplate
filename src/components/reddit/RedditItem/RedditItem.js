import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { removeItem } from '../../../actions/redditActions';
import './RedditItem.scss';

export class RedditItem extends React.Component {
	constructor() {
		super();
		this.handleDelete = this.handleDelete.bind(this);
	}

	handleDelete() {
		const { dispatch, index } = this.props;
		dispatch(removeItem(index));
	}

	render() {
		return (
			<div className="RedditItem">
				<h3>{ this.props.children }</h3>
				<p>Author: {this.props.author}</p>
				<button
					onClick={this.handleDelete}
				>
          X
				</button>
			</div>
		);
	}
}

RedditItem.propTypes = {
	author: PropTypes.string.isRequired,
	index: PropTypes.number.isRequired,
	dispatch: PropTypes.func.isRequired,
	children: PropTypes.string,
};

RedditItem.defaultProps = {
	redditData: [],
	author: 'Default',
	children: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorum dolorem architecto eos, esse facere quo omnis expedita, impedit officiis! Quisquam officiis recusandae expedita numquam vel in molestias qui placeat esse.',
};

export default connect(null, null)(RedditItem);
