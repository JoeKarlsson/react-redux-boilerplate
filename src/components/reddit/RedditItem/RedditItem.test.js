import React from 'react';
import renderer from 'react-test-renderer';
import { RedditItem } from './RedditItem';

test('RedditItem page snapshot matches', () => {
	const component = renderer.create(
		<RedditItem index={0} author="Joe Karlsson" />,
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
