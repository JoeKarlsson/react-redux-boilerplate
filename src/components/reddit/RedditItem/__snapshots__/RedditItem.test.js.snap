// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedditItem page snapshot matches 1`] = `
<div
  className="RedditItem"
>
  <h3>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorum dolorem architecto eos, esse facere quo omnis expedita, impedit officiis! Quisquam officiis recusandae expedita numquam vel in molestias qui placeat esse.
  </h3>
  <p>
    Author: 
    Joe Karlsson
  </p>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
